### PR TITLE
ASM-7793 Fix logic error in comparing CompCU versions

### PR DESCRIPTION
**Pull request by gavin-scott**
Previous code was treating 6.7 as newer than 7.0.1

### CODE REVIEW

**Comment by gavin-scott** [comment]
Fix for #10 

@mdcondon can you review since I screwed this up so embarrasingly on the last PR?  Thanks!

File: build.gradle [hunk 1]
```diff
@@ -76,11 +76,38 @@ class CompCuInfo {
...
+        // / Check micro
+        if (micro > o.micro) {
+            return true;
+        } else if (micro < o.micro) {
+            return false;
+        }
+
+        // Same version
+        return false;
+    }
```
Comment (hribab) [inline build.gradle]: [major, minor, micro].join().toInteger() > [o.major, o.minor, o.micro].join().toInteger()

Comment (gavin-scott) [inline build.gradle]: I don't think that works for e.g. 1.10 vs. 2.1

**Comment by gavin-scott** [comment]
@sushilrai or @ripienaar can you have a look at this?

**Comment by ripienaar** [comment]
Logic seems fine, I don't really know Groovy but if there's nothing built in and we dont want to get a extra library then this seems fine to me, @sushilrai any feedback?

**Comment by sushilrai** [comment]
Haven't worked on build.gradle before. @gavin-scott @mdcondon can confirm on the changes

**Comment by gavin-scott** [comment]
Merging based on the few comments and fact that no one has much groovy expertise since this fix is needed.

**Pull request merged**